workshop/3 - Cost Optimization/README.md

```diff
@@ -0,0 +1,260 @@
+    ```
+
+1. Now let's provision the Azure resources and deploy the sample application. In the PowerShell terminal, make sure you're still in the **azd-sample** directory and run the following command:
+
+    ```powershell
+    azd up
+    ```
+
+    You will be prompted to pick a subscription and Azure region. Select your subscription option, and pick **EastUS** for the region.
+
+2. When the provisioning and deployment is finished, the URL for the sample application will be displayed in the terminal. Open that URL in a browser. You should see the sample application running.
+
+3. You can also view the Azure resources in the Azure portal. Open the portal and browse for resource groups. The name of the resource group will start with  **<USERNAME>-cost-rg**.
+
+4. Open the App Service Plan within the resource group. You should see that it is provisioned with a **B1** SKU.
+
+    ![Screenshot of the App Service Plan](../images/3-Cost%20Optimization/app-service-plan.png)
+
+## Aim for scalable costs
+
+You want to have your resources scale up and down as needed. Azure App Service has built-in autoscaling capabilities. Let's add that to our sample application.
+
+1. In Visual Studio or VS Code, open the **autoscale.bicep** file from the **azd-sample\infra** folder.
+1. Note how it defines scaling rules both with a trigger and an action to take when the trigger is met. For example, if the CPU percentage is greater than a threshold, increase the number of instances by 1.
+1. Let's implement the autoscaling as part of our azd provisioning. Add the following to the **resources.bicep** file.
+
+    ```bicep
+    module webAppAutoScale 'autoscale.bicep' = {
+        name: 'deploy-${webAppServicePlan.name}-autoscale'
+        params: {
+        appServicePlanName: webAppServicePlan.name
+        location: location
+        isProd: isProd
+        tags: tags
+        }
+    }
+    ```
+
+1. In a PowerShell terminal, in the **\azd-sample** directory, run the following to update the azd environment.
+
+    ```powershell
+    azd env set IS_PROD true
+    ```
+
+1. Now, run the following to provision the autoscaling.
+
+    ```powershell
+    azd provision
+    ```
+
+1. Open up the same App Service Plan as before in the Azure portal (or just refresh your browser if you already have it open). You should see that it is provisioned with a **P1v3** SKU.
+
+    ![Screenshot of the App Service Plan](../images/3-Cost%20Optimization/app-service-plan-p1v3.png)
+
+1. Click on **Scale out (App Service Plan)** in the left navigation menu, under Settings. You should see that autoscaling is enabled with a "Rules Based" scale out method.  Click on **Manage rules based scaling** to see the scaling rules.
+
+    ![Screenshot of the scale out settings](../images/3-Cost%20Optimization/scale-out.png)
+
+> Note: Due to limits of the free subsription, the autoscaling may not work. However, it will work if you have a paid subscription.
```

Comment:
```suggestion
> Note: Due to limits of the free subscription, the autoscaling may not work. However, it will work if you have a paid subscription.
```